pkg/model/config/config.go

```diff
@@ -147,34 +147,53 @@ func (c Config) resourceAPIVersionCompatible(verType, version string) bool {
 	return true
 }
 
-// GVK contains information about scaffolded resources
-type GVK struct {
+// Data contains information about scaffolded resources
+type Data struct {
 	Group   string `json:"group,omitempty"`
 	Version string `json:"version,omitempty"`
 	Kind    string `json:"kind,omitempty"`
 
-	// CRDVersion holds the CustomResourceDefinition API version used for the GVK.
+	// CRDVersion holds the CustomResourceDefinition API version used for the Data.
 	CRDVersion string `json:"crdVersion,omitempty"`
-	// WebhookVersion holds the {Validating,Mutating}WebhookConfiguration API version used for the GVK.
+	// WebhookVersion holds the {Validating,Mutating}WebhookConfiguration API version used for the Data.
 	WebhookVersion string `json:"webhookVersion,omitempty"`
+
+	// Webhooks holds the Webhooks data
+	Webhooks Webhooks `json:"webhooks,omitempty"`
+}
+
+// Webhooks contains information about scaffolded webhooks
+type Webhooks struct {
+	Defaulting bool `json:"defaulting,omitempty"`
+	Validation bool `json:"validation,omitempty"`
+	Conversion bool `json:"conversion,omitempty"`
 }
 
-// isEqualTo compares it with another resource
-func (r GVK) isEqualTo(other GVK) bool {
+// isGVKEqualTo compares it with another resource
+func (r *Data) isGVKEqualTo(other Data) bool {
 	return r.Group == other.Group &&
 		r.Version == other.Version &&
 		r.Kind == other.Kind
 }
 
 // merge combines fields of two GVKs that have matching group, version, and kind,
 // favoring the receiver's values.
-func (r *GVK) merge(other GVK) {
+func (r *Data) merge(other Data) {
 	if r.CRDVersion == "" && other.CRDVersion != "" {
 		r.CRDVersion = other.CRDVersion
 	}
 	if r.WebhookVersion == "" && other.WebhookVersion != "" {
 		r.WebhookVersion = other.WebhookVersion
 	}
+	r.Webhooks.merge(other.Webhooks)
```

Comment:
This also needs to be updated.

Reply:
Done with caveats but to achieve same 👍  

```go
if other.Webhooks != nil {
	r.Webhooks.merge(*other.Webhooks)
}
```

And then, centralize the logic in the merge instead of it: 

```go
func (w *Webhooks) merge(other Webhooks) {
	if w == nil {
		w = &other
		return
	}

```